src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+interface Props {
+  onSelect?: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelect = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuary, setAppliedQuery] = useState('');
+  const [showList, setShowList] = useState(false);
+
+  const appliQuary = useCallback(debounce(setAppliedQuery, delay), [
+    setAppliedQuery,
+  ]);
+
+  const filteredLists = useMemo(() => {
+    return peopleFromServer.filter(people =>
+      people.name.toLowerCase().includes(appliedQuary.toLowerCase()),
+    );
+  }, [appliedQuary]);
+
+  const onselectedPerson = (person: Person) => {
+    setQuery(person?.name);
+    onSelect(person);
+    setShowList(false);
+  };
+
+  const handleQuaryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    appliQuary(event.target.value);
+    setShowList(true);
+    onSelect(null);
+  };
+
+  return (
+    <div className={cn('dropdown', { 'is-active': showList })}>
+      <div className="dropdown-trigger">
+        <input
+          value={query}
+          onChange={handleQuaryChange}
+          onFocus={() => setShowList(true)}
+          onBlur={event => {
+            if (!event.relatedTarget) {
+              setShowList(false);
+            }
+          }}
```

Comment:
Pls move the function inside `onBlur` handler to a separate const and use as a callback here